core/src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -200,43 +200,29 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
   def referencedTiColumns(expression: TiExpression): Seq[TiColumnRef] =
     PredicateUtils.extractColumnRefFromExpression(expression).asScala.toSeq
 
-  def extractTiColumnRefFromExpression(expression: TiExpression): mutable.HashSet[TiColumnRef] = {
-    val set: mutable.HashSet[TiColumnRef] = mutable.HashSet.empty[TiColumnRef]
-    expression match {
-      case r: TiColumnRef => set += r
-      case _: Constant    =>
-      case e: TiExpression =>
-        for (child <- e.getChildren.asScala) {
-          extractTiColumnRefFromExpression(child).foreach { set += _ }
-        }
-    }
-    set
-  }
-
-  def extractTiColumnRefFromExpressions(expressions: Seq[TiExpression]): Seq[TiColumnRef] = {
-    val set: mutable.HashSet[TiColumnRef] = mutable.HashSet.empty[TiColumnRef]
-    for (expression <- expressions) {
-      extractTiColumnRefFromExpression(expression).foreach { set += _ }
-    }
-    set.toSeq
-  }
-
   /**
-   * build a Seq of used TiColumnRef from AttributeSet and bound them to souce table
+   * build a Seq of used TiColumnRef from AttributeSet and bound them to source table
    *
    * @param attributeSet AttributeSet containing projects w/ or w/o filters
    * @param source source TiDBRelation
    * @return a Seq of TiColumnRef extracted
    */
   def buildTiColumnRefFromColumnSet(attributeSet: AttributeSet,
                                     source: TiDBRelation): Seq[TiColumnRef] = {
-    val tiColumnSet: Seq[TiExpression] = attributeSet.toSeq.collect {
+    val tiColumnSeq: Seq[TiExpression] = attributeSet.toSeq.collect {
       case BasicExpression(expr) => expr
     }
     val resolver = new MetaResolver(source.table)
-    val tiColumns: Seq[TiColumnRef] = extractTiColumnRefFromExpressions(tiColumnSet)
+    var tiColumns: mutable.HashSet[TiColumnRef] = mutable.HashSet.empty[TiColumnRef]
+    for (expression <- tiColumnSeq) {
+      val colSetPerExpr = PredicateUtils
+        .extractColumnRefFromExpression(expression.asInstanceOf[com.pingcap.tikv.expression.Expression])
```

Comment:
`.asInstanceOf[com.pingcap.tikv.expression.Expression]` is useless

Reply:
DONE.